mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/scu35ea-AI.lua

```diff
@@ -0,0 +1,120 @@
+else
+AlliedArmorTypes = { "1tnk", "2tnk" }
+end
+if Map.LobbyOption("difficulty") == "hard" then
+  AlliedNavyGuard = { "ca", "ca" }
+else
+  AlliedNavyGuard = { "ca" }
+end
+AlliedWarFactRally = { waypoint2, waypoint9, waypoint10, waypoint11 }
+InfAttack = { }
+ArmorAttack = { }
+
+SendAttackToBase = function(units)
+  Utils.Do(units, function(unit)
+    if not unit.IsDead and unit.HasProperty("Hunt") then
+      unit.AttackMove(waypoint77.Location, 2)
+      Trigger.OnIdle(unit, unit.Hunt)
+    end
+  end)
+end
+
+UnitsJustHunt = function(units)
+  Utils.Do(units, function(unit)
+    if not unit.IsDead and unit.HasProperty("Hunt") then
+      Trigger.OnIdle(unit, unit.Hunt)
+    end
+  end)
+end
+
+ProduceInfantry = function()
+	if AlliedBarracks01.IsDead then
+		return
+	elseif (OreTruck01.IsDead and OreTruck02.IsDead) and greece.Resources <= 299 then
+		return
+	end
+
+	local delay = Utils.RandomInteger(DateTime.Seconds(1), DateTime.Seconds(2))
+	local toBuild = { Utils.Random(AlliedInfantryTypes) }
+	greece.Build(toBuild, function(unit)
+		InfAttack[#InfAttack + 1] = unit[1]
+
+		if #InfAttack >= 5 then
+			UnitsJustHunt(InfAttack)
+			InfAttack = { }
+			Trigger.AfterDelay(DateTime.Seconds(1), ProduceInfantry)
+		else
+			Trigger.AfterDelay(delay, ProduceInfantry)
+		end
+	end)
+end
+
+ProduceArmor = function()
+	if AlliedWarFact01.IsDead and AlliedWarFact02.IsDead then
+		return
+	elseif (OreTruck01.IsDead and OreTruck02.IsDead) and greece.Resources <= 699 then
+		return
+	end
+
+	local delay = Utils.RandomInteger(DateTime.Seconds(7), DateTime.Seconds(10))
+	local toBuild = { Utils.Random(AlliedArmorTypes) }
+  local Rally = Utils.Random(AlliedWarFactRally)
+  Utils.Do(AlliedWarFact, function(fact)
+    fact.RallyPoint = Rally.Location
+  end)
+	greece.Build(toBuild, function(unit)
+		ArmorAttack[#ArmorAttack + 1] = unit[1]
+    
+    if Map.LobbyOption("difficulty") == "easy" then
+      if #ArmorAttack >= 2 then
+        SendAttackToBase(ArmorAttack)
+        ArmorAttack = { }
+        Trigger.AfterDelay(DateTime.Seconds(45), ProduceArmor)
+      else
+        Trigger.AfterDelay(delay, ProduceArmor)
+      end
+    elseif Map.LobbyOption("difficulty") == "normal" then
+      if #ArmorAttack >= 5 then
+        SendAttackToBase(ArmorAttack)
+        ArmorAttack = { }
+        Trigger.AfterDelay(DateTime.Seconds(30), ProduceArmor)
+      else
+        Trigger.AfterDelay(delay, ProduceArmor)
+      end
+    elseif Map.LobbyOption("difficulty") == "hard" then
+      if #ArmorAttack >= 8 then
+        SendAttackToBase(ArmorAttack)
+        ArmorAttack = { }
+        Trigger.AfterDelay(DateTime.Seconds(10), ProduceArmor)
+      else
+        Trigger.AfterDelay(delay, ProduceArmor)
+      end
+    end
+	end)
+end
+
+ProduceNavyGuard = function()
+	if NavalYard01.IsDead then
+		return
+	elseif (OreTruck01.IsDead and OreTruck02.IsDead) and greece.Resources <= 2399 then
+		return
+	end
+  NavalYard01.RallyPoint = waypoint26.Location
```

Comment:
You seem to be using a mix of tabs and spaces for indention. Please just use tabs.